### ISSUE TITLE
leave alt empty when no alt text provided

### ISSUE THREAD

**Issue by macmanx2** [MLTSHP/best-of-mltshp-bot#8]
If there is no alt text provided on the MLTSHP image, it's currently shared to Mastodon with "No alt provided," which could be seen as very much not good by some people (Mastodon in general loves alt text, so do I FWIW).

If there is no alt text provided on the MLTSHP image, could the bot simply not include alt text with the share?

**Comment by dphiffer** [comment]
I'm ok with replacing the phrase `No alt provided` with an empty string.

**Comment by dphiffer** [comment]
See also: https://mastodon.online/@NatureMC/112496203991377396

**Comment by jessamynwest** [comment]
What about something like "Alt in original image" I just want to be clear that we are 100% not just ignoring that alt text is a thing.

**Comment by macmanx2** [comment]
Oh, we should definitely include alt text when available, and the bot is doing that.

But, when there is not alt text available, like https://mltshp.com/p/1Q2MR it's coming in to Mastodon as https://mefi.social/@best_of_mltshp/112497594059051753 with:

`alt="No alt provided"`

**Comment by jessamynwest** [comment]
Ah I hear you. Tricky issue. I feel like leaving it blank it more user hostile than "no alt provided" but I don't really know what the visually disabled community thinks about that.

**Comment by macmanx2** [comment]
Yeah, I hear ya. I feel like "no alt text provided" gives the impression that someone actually typed that in, like the replier at https://mastodon.online/@NatureMC/112496203991377396 felt.

**Comment by dphiffer** [comment]
I can't decide which of these two statements I prefer:
- This is really a question about how we handle missing alt at mltshp.com, so we would be better off adjusting things on that code base if at all.
- Mastodon is a different community, and we should comport ourselves differently when in the company of fediverse strangers.

I can see it both ways tbh.

**Comment by jessamynwest** [comment]
I reached out to the user who was complaining. Not sure what might work better but we definitely have some options.

**Comment by dphiffer** [comment]
Ooh, I like the idea of having the bot follow an alt text reminder bot.

**Comment by jessamynwest** [comment]
That might be the simplest way to handle it though it might make me get all surly with non-alt-text-providing MLTSHPers but that's a separate issue.

**Comment by jessamynwest** [comment]
We got some feedback (and a bit of an apology) which says we could add a hashtag that might get a bot to do a thing? So you can add  #alt4me I don't know if this could be added only when there's no or blank alt but worth thinking about.

https://mastodon.online/@NatureMC/112500711922416627

**Comment by dphiffer** [comment]
Proposal: if the alt is set to `No alt provided` post with `alt=""` and the hashtag `#alt4me`, and one of us signed in as the bot updates manually (this part could be further improved, as a follow-on enhancement).

**Comment by dphiffer** [comment]
Seems like this is working!
